python/plugins/grassprovider/ext/r_watershed.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    r_watershed.py
+    --------
+    Date                 : June 2021
+    Copyright            : (C) 2021 by Nicolas Godet
+    Email                : nicolas dot godet at outlook dot fr
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Nicolas Godet'
+__date__ = 'June 2021'
+__copyright__ = '(C) 2021, Nicolas Godet'
+
+
+def checkParameterValuesBeforeExecuting(alg, parameters, context):
```

Comment:
Instead of this custom code, just hardcoded the minimum value in the parameter definition line in the txt file

Reply:
I wish I could.
If I hardcode the minimum value to 1 and leave default value to None, the tool will initialize with the minimum value.

Reply:
Hmmmm that's a bug then -- it should be possible to do this.

Reply:
Should I report this ?

Reply:
> 
> 
> I wish I could.
> If I hardcode the minimum value to 1 and leave default value to None, the tool will initialize with the minimum value.


it isn't just enough something like

`QgsProcessingParameterNumber|threshold|Minimum size of exterior watershed basin|QgsProcessingParameterNumber.Integer|1000|True|None|None`

it seems to work as expected here

`r.watershed elevation=rast_60dc4b8e51d5f2 threshold=1000 convergence=5 memory=300 accumulation=accumulationc03d19ee537c4382bee72e3393a93458 drainage=drainagec03d19ee537c4382bee72e3393a93458 basin=basinc03d19ee537c4382bee72e3393a93458 stream=streamc03d19ee537c4382bee72e3393a93458 half_basin=half_basinc03d19ee537c4382bee72e3393a93458 length_slope=length_slopec03d19ee537c4382bee72e3393a93458 slope_steepness=slope_steepnessc03d19ee537c4382bee72e3393a93458 tci=tcic03d19ee537c4382bee72e3393a93458 spi=spic03d19ee537c4382bee72e3393a93458 --overwrite
`

Reply:
The goal is to leave the default value blank but impose a minimum value. Nyall would like to force the user to set this value manually.

If you set `QgsProcessingParameterNumber|threshold|Minimum size of exterior watershed basin|QgsProcessingParameterNumber.Integer|None|True|1|None`, it will default to 1 which is, apparently, not the expected behavior.